### ISSUE TITLE
[V2] [IntlProvider] How to change locale and messages at runtime?

### ISSUE THREAD

**Issue by BerndWessels** [yahoo/react-intl#281]
Hello
I have wrapped my App as proposed for V2 with the `<IntlProvider>` like this:

    import {IntlProvider, addLocaleData} from 'react-intl';
    import intlDE from 'react-intl/lib/locale-data/de';
    import intlEN from 'react-intl/lib/locale-data/en';
    import intlMessagesDE from '../public/assets/translations/de.json';
    import intlMessagesEN from '../public/assets/translations/en.json';

    addLocaleData(intlDE);
    addLocaleData(intlEN);

    ReactDOM.render(
      <IntlProvider locale="de" messages={intlMessagesDE}>
        <RelayRouter
          history={browserHistory}
          routes={routes}
        />
      </IntlProvider>,
      document.getElementById('root')
    );

But I can't figure out how to change the `locale` and the `messages` at runtime.

Basically I want to allow the user to change the language by selecting it from a `<select>` dropdown. That should change the `locale` and the `messages` to the selected language and update the UI.

Can this be done without a full page reload?

Thanks
Bernd

**Comment by ericf** [comment]
It seems you've asked this question twice and someone responded in the other place.

**Comment by dolyst** [comment]
Any chance of a link to the duplicate question please?  I'm interested in the answer.  Many thanks!

**Comment by mlsogaard** [comment]
@dolyst I guess not!
